Bug: Flaky test

Failure in CI:

```
______________________________ test_file_response ______________________________

tmpdir = local('/tmp/pytest-of-runner/pytest-0/test_file_response0')

    def test_file_response(tmpdir: Path) -> None:
        date_string = formatdate(datetime.now().timestamp(), usegmt=True)
        path = tmpdir / "xyz"
        content = b"<file content>" * 1000
        Path(path).write_bytes(content)
    
        filled_by_bg_task = ""
    
        async def numbers(minimum: int, maximum: int) -> AsyncIterator[str]:
            for i in range(minimum, maximum + 1):
                yield str(i)
                if i != maximum:
                    yield ", "
                await anyio.sleep(0)
    
        async def numbers_for_cleanup(start: int = 1, stop: int = 5) -> None:
            nonlocal filled_by_bg_task
            async for thing in numbers(start, stop):
                filled_by_bg_task = filled_by_bg_task + thing
    
        cleanup_task = BackgroundTask(numbers_for_cleanup, start=6, stop=9)
    
        async def app(scope: "Scope", receive: "Receive", send: "Send") -> None:
            response = FileResponse(path=path, filename="example.png", background=cleanup_task)
            await response(scope, receive, send)
    
        assert filled_by_bg_task == ""
        client = TestClient(app)
        response = client.get("/")
        expected_disposition = 'attachment; filename="example.png"'
        assert response.status_code == HTTP_200_OK
        assert response.content == content
        assert response.headers["content-type"] == "image/png"
        assert response.headers["content-disposition"] == expected_disposition
        assert response.headers["content-length"] == "14000"
>       assert response.headers["last-modified"].lower() == date_string.lower()
E       AssertionError: assert 'fri, 28 oct ... 12:30:23 gmt' == 'fri, 28 oct ... 12:30:24 gmt'
E         - fri, 28 oct 2022 12:30:24 gmt
E         ?                         ^
E         + fri, 28 oct 2022 12:30:23 gmt
E         ?                         ^

tests/response/test_file_response.py:58: AssertionError
=========================== short test summary info ============================
FAILED tests/response/test_file_response.py::test_file_response - AssertionError: assert 'fri, 28 oct ... 12:30:23 gmt' == 'fri, 28 oct ... 12:30:24 gmt'
  - fri, 28 oct 2022 12:30:24 gmt
  ?                         ^
  + fri, 28 oct 2022 12:30:23 gmt
  ?                         ^
============ 1 failed, 1130 passed, 5 skipped, 1 xfailed in 19.12s =============
```

--- a/tests/response/test_file_response.py
+++ b/tests/response/test_file_response.py
@@ -3,7 +3,6 @@
 https://github.com/encode/starlette/blob/master/tests/test_responses.py And are
 meant to ensure our compatibility with their API.
 """
-from datetime import datetime
 from email.utils import formatdate
 from pathlib import Path
 from typing import TYPE_CHECKING, AsyncIterator
@@ -21,10 +20,10 @@ if TYPE_CHECKING:
 
 
 def test_file_response(tmpdir: Path) -> None:
-    date_string = formatdate(datetime.now().timestamp(), usegmt=True)
     path = tmpdir / "xyz"
     content = b"<file content>" * 1000
     Path(path).write_bytes(content)
+    date_string = formatdate(Path(path).stat().st_mtime, usegmt=True)
 
     filled_by_bg_task = ""
 

